code/modules/reagents/dispenser/dispenser2.dm

```diff
@@ -35,6 +39,7 @@
 	to_chat(user, "It has [cartridges.len] cartridges installed, and has space for [DISPENSER_MAX_CARTRIDGES - cartridges.len] more.")
 
 /obj/machinery/chemical_dispenser/proc/add_cartridge(obj/item/chems/chem_disp_cartridge/C, mob/user)
+	. = FALSE
```

Comment:
On line 33-35, you aren't allowed to add full cartridges if it's being manually built. Check the `full_populate` var. You may also need to dump cartridges in deconstruct if it doesn't do so already.

Reply:
Assuming you meant ``populate_parts`` here, but should be resolved